lib/Conversion/ImportVerilog/Statements.cpp

```diff
@@ -642,6 +642,36 @@ struct StmtVisitor {
     return success();
   }
 
+  // Handle concurrent assertion statements.
+  LogicalResult visit(const slang::ast::ConcurrentAssertionStatement &stmt) {
+    auto loc = context.convertLocation(stmt.sourceRange);
+    auto property = context.convertAssertionExpression(stmt.propertySpec, loc);
+    if (!property)
+      return failure();
+
+    // Handle assertion statements that don't have an action block.
+    if (stmt.ifTrue && stmt.ifTrue->as_if<slang::ast::EmptyStatement>()) {
+      switch (stmt.assertionKind) {
+      case slang::ast::AssertionKind::Assert:
+        builder.create<verif::AssertOp>(loc, property, Value(), StringAttr{});
+        return success();
+      case slang::ast::AssertionKind::Assume:
+        builder.create<verif::AssumeOp>(loc, property, Value(), StringAttr{});
+        return success();
+      default:
+        break;
+      }
+      mlir::emitError(loc) << "unsupported concurrent assertion kind: "
+                           << slang::ast::toString(stmt.assertionKind);
+      return failure();
+    }
+
+    mlir::emitError(loc)
+        << "concurrent assertion statements with action blocks "
+           "are not supported yet";
+    return failure();
```

Comment:
Excellent idea to leave this for a later PR :+1:

Reply:
I think the big question is going to be where you would place the IR ops for the stuff in the action blocks. You can have basically arbitrary SV code in the action blocks, so these would probably have to get lowered into some procedural region. But `verif.assert` has no such procedural region, and there's also no Verif op that can check an LTL property and then execute a region of code. So this might actually require a new IR operation to handle -- which isn't a bad thing.

Reply:
Somehow related to this I did not want to handle `SequenceWithMatchExpr` as I think this will also involve more IR ops.

Reply:
Yeah good idea!